lock/src/main/java/org/infinispan/lock/api/impl/manager/EmbeddedClusteredLockManager.java

```diff
@@ -0,0 +1,152 @@
+package org.infinispan.lock.api.impl.manager;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.lock.api.ClusteredLock;
+import org.infinispan.lock.api.ClusteredLockConfiguration;
+import org.infinispan.lock.api.ClusteredLockManager;
+import org.infinispan.lock.api.impl.entries.ClusteredLockKey;
+import org.infinispan.lock.api.impl.entries.ClusteredLockState;
+import org.infinispan.lock.api.impl.entries.ClusteredLockValue;
+import org.infinispan.lock.api.impl.lock.ClusteredLockImpl;
+import org.infinispan.lock.api.impl.log.Log;
+import org.infinispan.lock.exception.ClusteredLockException;
+import org.infinispan.util.ByteString;
+
+/**
+ * The Embedded version for the lock cluster manager
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 9.2
+ */
+public class EmbeddedClusteredLockManager implements ClusteredLockManager {
+
+   private static final long WAIT_CACHES_TIMEOUT = TimeUnit.SECONDS.toNanos(15);
+   private static final Log log = LogFactory.getLog(EmbeddedClusteredLockManager.class, Log.class);
+
+   private CompletableFuture<CacheHolder> cacheHolderFuture;
+   private ConcurrentHashMap<String, ClusteredLock> locks = new ConcurrentHashMap();
+   private ScheduledExecutorService scheduledExecutorService;
+   private Executor executor;
+
+   private AdvancedCache<ClusteredLockKey, ClusteredLockValue> cache;
+
+   public EmbeddedClusteredLockManager(CompletableFuture<CacheHolder> cacheHolderFuture) {
+      this.cacheHolderFuture = cacheHolderFuture;
+   }
+
+   @Inject
+   public void injectDep(@ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR) ScheduledExecutorService scheduledExecutorService,
+                         @ComponentName(KnownComponentNames.ASYNC_OPERATIONS_EXECUTOR) Executor executor) {
+      this.scheduledExecutorService = scheduledExecutorService;
+      this.executor = executor;
+   }
+
+   @Override
+   public boolean defineLock(String name) {
+      return defineLock(name, new ClusteredLockConfiguration());
+   }
+
+   @Override
+   public boolean defineLock(String name, ClusteredLockConfiguration configuration) {
+      // TODO: Configuration is not used because we don't support any other mode for now. For that : ISPN-8413
+      CacheHolder cacheHolder = extractCacheHolder(cacheHolderFuture);
+      cache = cacheHolder.getClusteredLockCache();
```

Comment:
can throw `NullPointerException`

Reply:
if the cacheHolder is null you mean ?

Reply:
Ok, I'm throwing an exception if the caches could not be started instead of returning null